mozetl/taar/taar_similarity.py

```diff
@@ -0,0 +1,163 @@
+"""
+Bug 1386274 - TAAR similarity-based addon donor list
+
+This job clusters users into different groups based on their
+active addons. A representative users sample is selected from
+each cluster ("donors") and is saved to a model file along
+with a feature vector that will be used, by the TAAR library
+module, to perform recommendations.
+"""
+
+import click
+import json
+import logging
+
+from pyspark.sql import SparkSession
+from pyspark.ml.feature import HashingTF, IDF
+from pyspark.ml.clustering import BisectingKMeans
+from pyspark.ml import Pipeline
+from taar_utils import store_json_to_s3
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def get_samples(spark):
+    """ Get a dataframe with a valid set of sample to base the next
+    processing on.
+    """
+    return spark.sql("SELECT * FROM longitudinal")\
+        .where("active_addons IS NOT null")\
+        .where("size(active_addons[0]) > 2")\
+        .where("size(active_addons[0]) < 100")\
+        .where("normalized_channel = 'release'")\
```

Comment:
Don't we also want:
` AND build IS NOT NULL AND build[0].application_name='Firefox'`

Reply:
Good catch!